src/app/patient-summary/components/PatientSummary.tsx

```diff
@@ -14,11 +16,16 @@ export default function PatientSummary({ fhirBundle }) {
   const [selectedTab, setSelectedTab] = useState(dataTabs[0] || '');
   const renderPanels = () =>
     dataTabs.map((resourceType) => {
+      const resourceInfo = extractResourceInfo(
+        ResourceType[resourceType],
+        fhirBundle,
+      );
+
       return (
         <TabPanel value={resourceType} index={selectedTab} key={resourceType}>
           {/* TODO: Consider enhancing the conditional rendering of resources */}
           {resourceType === 'Patient' ? (
-            <Patient data={fhirBundle} />
+            <Patient patientInfo={resourceInfo} />
```

Comment:
you continue using data as the param name. This way you can use a look up that returns a component based on the resource type and all the components take the same param "data"

you could have something like 
getComponent =(resourceType, data) => {
   const componentMap = { 'Patient': Patient };
   const TabComponent = componentMap[resourceType];

   return <TabComponent data={data} />
   }